projects/libyuv/libyuv_fuzzer.cc

```diff
@@ -0,0 +1,80 @@
+/* # Copyright 2020 Google Inc. */
+/* # */
+/* # Licensed under the Apache License, Version 2.0 (the "License"); */
+/* # you may not use this file except in compliance with the License. */
+/* # You may obtain a copy of the License at */
+/* # */
+/* #      http://www.apache.org/licenses/LICENSE-2.0 */
+/* # */
+/* # Unless required by applicable law or agreed to in writing, software */
+/* # distributed under the License is distributed on an "AS IS" BASIS, */
+/* # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. */
+/* # See the License for the specific language governing permissions and */
+/* # limitations under the License. */
+/* # */
+/* ################################################################################ */
+
+#include <iostream>
+#include <vector>
+#include <cmath>
+
+#include <fuzzer/FuzzedDataProvider.h>
+#include "libyuv.h"
+#include "fuzz_common.h"
+
+using namespace std;
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+
+    FuzzedDataProvider fuzzed_data(data, size);
+
+    int range_size = sqrt((static_cast<int>(size) + 1) / 2);
+    if (range_size == 0)
+    {
+        libyuv::I420ToNV21(data, 0, data, 0, data, 0, nullptr, 0, nullptr, 0, 0, 0);
+        return 0;
+    }
+
+    size_t width = fuzzed_data.ConsumeIntegralInRange(1, range_size);
+    size_t height = fuzzed_data.ConsumeIntegralInRange(1, range_size);
+    bool cpu_flag = fuzzed_data.ConsumeBool();
+
+    write_conf(width, height, cpu_flag);
+
+    if (cpu_flag == 1)
+    {
+        //enable all cpu specific optimizations.
+        libyuv::MaskCpuFlags(-1);
```

Comment:
this only tests 1 instruction set - whatever the host machine has.  e.g. avx2.
there is code for ssse3 and avx2 for most functions.  as well as c.

Reply:
Okay, got it.